server/webapp/WEB-INF/rails.new/app/helpers/api_v1/authentication_helper.rb

```diff
@@ -48,6 +48,15 @@ def check_admin_user_and_401
       end
     end
 
+    def check_pipeline_group_admin_user_and_401
+      groupName = params[:group] || go_config_service.findGroupNameByPipeline(com.thoughtworks.go.config.CaseInsensitiveString.new(params[:name]))
+      return unless security_service.isSecurityEnabled()
+      unless security_service.isUserAdminOfGroup(current_user.getUsername, groupName)
```

Comment:
Wouldn't this raise a null pointer exception if the group does not exist? Check out GoConfigService's function isUserAdminOfGroup()